frontend/src/pages/orders/OrderList.tsx

```diff
@@ -0,0 +1,118 @@
+import OrderService from "../../services/OrderService.ts";
+import React, {useEffect, useState} from "react";
+import {Order} from "../../types/Order.ts";
+import PageHeader from "../../layouts/PageHeader.tsx";
+import {Link} from "react-router-dom";
+
+const orderService = new OrderService();
+function OrderList() {
+    const [orders, setOrders] = useState<Order[]>([]);
+    const [filter, setFilter] = useState("");
+    const [ordersByStatus, setOrdersByStatus] = useState<Order[]>(orders);
+
+    const filteredOrders = ordersByStatus.filter(
+        (order) =>
+            order.customerName.toLowerCase().includes(filter.toLowerCase())
+    );
+
+    const handleStatusChange = (e: React.MouseEvent<HTMLInputElement>) => {
+        const target = e.target as HTMLElement;
+        if (target.id === "allOrders") {
+            setOrdersByStatus(orders)
+        }else {
+            setOrdersByStatus(orders.filter((order) => order.orderStatus === target.id));
+        }
+    };
+
+    useEffect(() => {
+        orderService.getAllOrders().then((response) => {
```

Comment:
I would suggest you, to write a seperate "getAllOrders" inside your OrderList.tsx file (not outsourced in your "OrderService.ts" file ). Also include a .catch() block in your axios request and print the error message message (with console.error or even better with a Logger, if you want).
E.g. add to line 17: 
`function getAllOrders()` 
`{axios.get('/api/orders')`
`.then((response) => {`
`setOrders(response.data);`
`setOrdersByStatus(response.data);`
`})`
`.catch((error) => {`
`Logger.error("An error occurred while loading all orders:", error);`
`});`

Reply:
I think defining axios operations in a separate class is a better use of object-oriented programming and SOLID.

I added catch block for error message.